File: app/src/main/java/dev/pitlor/smssync/viewmodels/SyncFragmentViewModel.java
package dev.pitlor.smssync.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.time.OffsetDateTime;

import dev.pitlor.smssync.datasources.AppDatabase;
import dev.pitlor.smssync.datasources.daos.SyncDao;

public class SyncFragmentViewModel extends BaseViewModel {
    public LiveData<OffsetDateTime> lastSync = appRepository.getLastSync();

    public SyncFragmentViewModel(@NonNull Application application) {
        super(application);
    }

    public boolean isLoading() {
        return lastSync == null;
    }
}

File: app/src/main/java/dev/pitlor/smssync/viewmodels/SyncFragmentViewModel.kt
package dev.pitlor.smssync.viewmodels

import android.app.Application
import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.LiveData
import dev.pitlor.smssync.datasources.AppRepository
import dev.pitlor.smssync.datasources.Sync
import java.time.OffsetDateTime

class SyncFragmentViewModel @ViewModelInject constructor(appRepository: AppRepository, application: Application) : BaseViewModel(application) {
    var lastSync = appRepository.getLastSync()
}